### ISSUE TITLE
There was an error in the prediction of T2SE.

### ISSUE THREAD

**Issue by YihangLin123** [zhangyumeng1sjtu/DeepSecE#3]
Hello, first of all, I would like to express my sincere gratitude for your excellent work! I would like to ask, I used the model checkpoint.pt you provided on GitHub to conduct the prediction experiment of effector proteins, and followed the commands in your README file. The results are shown in the following figure. The prediction of T2SE cannot reach the indicators in your paper. If there is any omission in your code or paper, please let me know. Thank you! I am looking forward to your reply very much!

![Image](https://github.com/user-attachments/assets/c3ae0fe4-abfa-470b-bedf-88c582711bf6)

I would also like to ask how you obtained the following table and how you used your model to test the five effector proteins to obtain the following indicators.

![Image](https://github.com/user-attachments/assets/17d8ce01-3988-4c34-85e7-8c68365afe4d)

**Comment by zhangyumeng1sjtu** [comment]
Hi Yihang,

We performed five-fold cross-validation and ensembled the five prediction scores to obtain the final predictions on the test set (as shown in the Jupyter notebook). But the checkpoint file provided on the website was derived from training a model on all the training data. I'm not sure what you mean by saying the prediction of T2SE cannot reach the indicators in the paper based on the figure you attached.

**Comment by YihangLin123** [comment]
Thank you very much for your prompt reply. Your response has been of great help to me! Thank you again for your excellent work!

I would also like to ask how you obtained the npy file used for drawing the chart?

If I have missed anything in the code you provided, I look forward to your guidance!

![Image](https://github.com/user-attachments/assets/802b3e8a-7fd2-4228-9bc5-ef1458b1a814)

**Comment by zhangyumeng1sjtu** [comment]
Hi Yihang,

I just saved the labels, predicted labels, and predicted probabilities in separate files using numpy in the predict function (not shown in the code). I hope it helps.

**Comment by YihangLin123** [comment]
Thank you very much for your prompt reply.

**Comment by YihangLin123** [comment]
I'm very sorry to bother you again. I followed your instructions and used the predict function to save the labels, predicted labels, and predicted probabilities as npy files using numpy. However, the results I obtained do not match the ones you provided, and I am unable to read and plot them. I hope you can provide the code for obtaining the npy files of valid and test. Thank you very much!
